var/spack/repos/builtin/packages/cosmomc/package.py

```diff
@@ -0,0 +1,192 @@
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+import fnmatch
+import os
+import shutil
+
+
+class Cosmomc(Package):
+    """CosmoMC is a Fortran 2008 Markov-Chain Monte-Carlo (MCMC) engine
+       for exploring cosmological parameter space, together with
+       Fortran and python code for analysing Monte-Carlo samples and
+       importance sampling (plus a suite of scripts for building grids
+       of runs, plotting and presenting results)."""
+
+    homepage = "http://cosmologist.info/cosmomc/"
+    url      = "https://github.com/cmbant/CosmoMC/archive/Nov2016.tar.gz"
+
+    version('Nov2016',  '98620cb746352f68fb0c1196e9a070ac', preferred=True)
+    version('June2016', '92dc651d1407cca6ea9228992165f5cb')
+
+    variant('mpi', default=True, description='Enable MPI support')
+    variant('planck', default=False,
+            description='Enable Planck Likelihood code and baseline data')
+
+    patch('Makefile.patch')
+    patch('errorstop.patch')
+
+    depends_on('mpi', when='+mpi')
+    depends_on('planck-likelihood', when='+planck')
+    depends_on('python @2.7:2.999,3.4:')
+
+    parallel = False
+
+    def install(self, spec, prefix):
+        # Clean up environment to avoid configure problems
+        os.environ.pop('LINKMPI', '')
+        os.environ.pop('NERSC_HOST', '')
+        os.environ.pop('NONCLIKLIKE', '')
+        os.environ.pop('PICO', '')
+        os.environ.pop('PRECISION', '')
+        os.environ.pop('RECOMBINATION', '')
+        os.environ.pop('WMAP', '')
+
+        # Set up Planck data if requested
+        clikdir = join_path('data', 'clik')
+        try:
+            os.remove(clikdir)
+        except OSError:
+            pass
+        if '+planck' in spec:
+            os.symlink(join_path(os.environ['CLIK_DATA'], 'plc_2.0'), clikdir)
+        else:
+            os.environ.pop('CLIK_DATA', '')
+            os.environ.pop('CLIK_PATH', '')
+            os.environ.pop('CLIK_PLUGIN', '')
+
+        # Choose compiler
+        # Note: Instead of checking the compiler vendor, we should
+        # rewrite the Makefile to use Spack's options all the time
+        if spec.compiler.name == 'gcc':
+            choosecomp = 'ifortErr=1'       # choose gfortran
+        elif spec.compiler.name == 'intel':
+            choosecomp = 'ifortErr=0'       # choose ifort
+        else:
+            raise InstallError("Only GCC and Intel compilers are supported")
+
+        # Configure MPI
+        if '+mpi' in spec:
+            wantmpi = 'BUILD=MPI'
+            mpif90 = 'MPIF90C=%s' % spec['mpi'].mpifc
+        else:
+            wantmpi = 'BUILD=NOMPI'
+            mpif90 = 'MPIF90C='
+        
+        # Choose BLAS and LAPACK
+        lapack = ("LAPACKL=%s %s" %
+                  (spec['lapack'].lapack_libs.ld_flags,
```

Comment:
do `spec['lapack'].lapack_libs+spec['blas'].blas_libs` and then ask `ld_flags` to automatically remove duplicates.